crates/matrix-sdk/src/event_cache/mod.rs

```diff
@@ -376,6 +410,34 @@ impl RoomEventCache {
 
         Ok((store.room_events(self.inner.room.room_id()).await?, self.inner.sender.subscribe()))
     }
+
+    /// Returns the earliest back-pagination token, that is, the one closest to
+    /// the beginning of the timeline as we know it.
+    ///
+    /// Optionally, wait at most for the given duration for a back-pagination
+    /// token to be returned by a sync.
+    pub async fn earliest_backpagination_token(
```

Comment:
Naming proposal: `most_recent_backpagination_token`?

Reply:
It's not the most recent, it's actually the "oldest" :smiling_face_with_tear: aka the closest to the start of the timeline. Should I rename it to `oldest_backpagination_token`, or even `furthest_backpagination_token`?

Reply:
Oh, then `oldest_backpagination_token` makes more sense to me!